src/content/blog/if-i-wrote-a-linter/index.mdx

```diff
@@ -0,0 +1,580 @@
+---
+pubDate: 2024-09-13
+description: "Why I'd write a TypeScript linter in TypeScript, build in TypeScript syntax and type awareness always, and other musings on the state of web linting in 2024."
+image:
+  alt: "Tevye from Fiddler on the Roof TODO_ACTIVITY"
+  src: "~/assets/blog/sisyphus.jpg"
+title: "If I Wrote a Linter"
+---
+
+import tevyeDancing from "~/assets/blog/tevye-dancing.webp";
+import LabeledImage from "~/components/blog/mdx/LabeledImage.astro";
+import LabeledVideo from "~/components/blog/mdx/LabeledVideo.astro";
+
+> # 🛑 **THIS IS JUST A DRAFT.**
+>
+> I need to run it by other developers in the linter ecosystem.
+> It might be horribly wrong and it might be horribly misrepresenting reality.
+> Please don't take it seriously!
+
+<LabeledImage
+	alt="Tevye from Fiddler on the Roof TODO_ACTIVITY"
+	description="Tevye from Fiddler on the Roof enjoying linting his code with a very strict configuration."
+	original="https://giphy.com/gifs/classic-movies-musicals-fiddler-on-the-roof-xULW8PseQFMgq8Bums"
+	src={tevyeDancing}
+/>
+
+Anybody who works with a project long enough inevitably fantasizes about rebuilding it themselves.
+Part of that is natural human _"Not Built Here"_ syndrome.
+Part is because any one person will have different drives and goals than the person or group of people in charge of the tool- even if they are one of them.
+And part is the inevitable struggle of long lived tools simultaneously trying to preserve legacy support and keep up with industry trends in real time.
+
+I've been working on TypeScript linting for a while.
+I started contributing to [TSLint](https://palantir.github.io/tslint) community projects in 2016 and am now a number of the [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) and [ESLint](https://github.com/eslint/eslint) teams.
+I enjoy both those projects.
+This post isn't to rebuttal of either project or their direction, just my idle fantasizing about what could be.
+
+## Core Architecture
+
+This is how I would choose to build a linter in 2024.
+
+### TypeScript Core
```

Comment:
I'm going to split this out into its own blog post. It's much more dev-friendly as a subject area. And I want it as its own standalone discussable (set of) point(s).

Reply:
#294